Hestia/Hestia/frontend/Local/UITableViewControllerServerConnect.cs

```diff
@@ -0,0 +1,124 @@
+using Foundation;
+using ObjCRuntime;
+using System;
+using UIKit;
+
+using Hestia.DevicesScreen.resources;
+using Hestia.backend;
+using Hestia.backend.models;
+using Hestia.backend.utils;
+
+namespace Hestia.DevicesScreen
+{
+    public partial class UITableViewControllerServerConnect : UITableViewController
+    {
+        NSUserDefaults userDefaults;
+
+        public UITableViewControllerServerConnect(IntPtr handle) : base(handle)
+        {
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+
+            newServerName.ShouldReturn += TextFieldShouldReturn;
+            newIP.ShouldReturn += TextFieldShouldReturn;
+            newPort.ShouldReturn += TextFieldShouldReturn;
+
+            newServerName.Tag = 1;
+            newIP.Tag = 2;
+            newPort.Tag = 3;
+
+            newServerName.BecomeFirstResponder();
+
+            userDefaults = NSUserDefaults.StandardUserDefaults;
+            var defaultServerName = userDefaults.StringForKey(Resources.strings.defaultsServerNameHestia);
+            if (defaultServerName != null)
+            {
+                newServerName.Text = defaultServerName;
+                newServerName.Placeholder = defaultServerName;
+            }
+
+            var defaultIP = userDefaults.StringForKey(Resources.strings.defaultsIpHestia);
+            if (defaultIP != null)
+            {
+                newIP.Text = defaultIP;
+                newIP.Placeholder = defaultIP;
+            }
+            var defaultPort = userDefaults.StringForKey(Resources.strings.defaultsPortHestia);
+            if (defaultPort != null)
+            {
+                newPort.Text = defaultPort;
+                newPort.Placeholder = defaultPort;
+            }
+        }
+
+        public override bool ShouldPerformSegue(string segueIdentifier, NSObject sender)
+        {
+            bool validIp = false;
+
+            try
+            {
+                validIp = PingServer.Check(newIP.Text, int.Parse(newPort.Text));
```

Comment:
As the customer said, just pinging the server isn't enough for checking if it's a valid Hestia server. Send a request instead

Reply:
We were aware of the bug, @Mikmist is looking into it. #140

Reply:
Alright, make sure this is fixed for the next sprint.